client/cmd/mmbot/main.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+/*
+Create core, wallets and register for server using dexc before starting the bot.
+*/
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/server/admin"
+
+	_ "decred.org/dcrdex/client/asset/btc"
+	_ "decred.org/dcrdex/client/asset/dcr"
+)
+
+var (
+	userDir, _            = os.UserHomeDir()
+	defaultConfigFileName = "mm.json"
+	defaultCoreDir        = filepath.Join(userDir, ".dexc")
+	defaultConfigFilePath = filepath.Join(defaultCoreDir, defaultConfigFileName)
+)
+
+func main() {
+	if err := mainErr(); err != nil {
+		fmt.Fprint(os.Stderr, err, "\n")
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func mainErr() error {
+	var (
+		c     *core.Core
+		pgmID uint64
+
+		// flags
+		configFile, coreDir, lang, logLevel string
+		testnet, simnet                     bool
+		manualRate                          float64
+	)
+
+	flag.StringVar(&configFile, "config", defaultConfigFilePath, "the bot program to run")
+	flag.StringVar(&coreDir, "coredir", defaultCoreDir, "the core configuration directory")
+	flag.StringVar(&lang, "lang", "en-US", "language")
+	flag.BoolVar(&testnet, "testnet", false, "use testnet")
+	flag.BoolVar(&simnet, "simnet", false, "use simnet")
+	flag.StringVar(&logLevel, "log", "info", "universal or per-logger log level. e.g. log=trace, or log=CORE=trace,BOT=trace")
+	flag.Float64Var(&manualRate, "emptyrate", 0, "(optional) a rate to use if the book is empty an there are no oracles available")
+	flag.Parse()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	loggerMaker, err := dex.NewLoggerMaker(os.Stdout, logLevel)
+	if err != nil {
+		return err
+	}
+	log := loggerMaker.NewLogger("MAIN")
+
+	killChan := make(chan os.Signal, 1)
+	signal.Notify(killChan, os.Interrupt)
+	go func() {
+		for range killChan {
+			if c != nil && pgmID > 0 {
+				if err := c.RetireBot(pgmID); err != nil {
+					fmt.Fprintf(os.Stderr, "error retiring bot: %v \n", err)
+				}
+
+			}
+			log.Infof("Shutdown signal received")
+			cancel()
```

Comment:
And `return` or `break`?  Do you want to allow subsequent interrupts to call RetireBot repeatedly?